src/views/Profile/PersonalInfo/PersonalInfo.tsx

```diff
@@ -16,40 +16,58 @@ import { useStyles } from './styles/PersonalInfo.styles';
 import { ErrorField } from './ErrorField';
 import { regex } from './constants/regex';
 import { fields } from './constants/fields';
-import { IErrorFields, INewAuthorValues, IVisitFields } from './types';
+import {
+  IEditAuthorProps,
+  IErrorFields,
+  INewAuthorValues,
+  IVisitFields,
+} from './types';
 import i18n, { langTokens } from '../../../locales/localizationInit';
 import RegionCityHandler from './RegionCityHandler';
 import { validation } from './constants/validation';
 import { validateInput } from './utilities/validateInput';
 
-export const PersonalInfo: React.FC = () => {
+export const PersonalInfo: React.FC<IEditAuthorProps> = ({ author }) => {
   const classes = useStyles();
 
   const [newAuthorValues, setNewAuthorValues] = useState<INewAuthorValues>({
-    avatar: '',
-    firstName: '',
-    lastName: '',
-    regionId: 0,
-    cityId: 0,
-    bio: '',
-    email: '',
-    socialNetwork: [null, null, null, null, null],
+    avatar: author ? author.avatar : '',
+    firstName: author ? author.firstName : '',
+    lastName: author ? author.lastName : '',
+    regionId: author ? author.region.id : null,
+    cityId: author ? author.mainInstitution.city.id : null,
+    bio: author ? author.bio : '',
+    email: author ? author.email : '',
+    socialNetwork: author
+      ? author.socialNetworks
+      : [null, null, null, null, null],
+    work: author ? author.mainInstitution.name : null,
   });
   const [errorFields, setErrorFields] = useState<IErrorFields>({
-    avatar: i18n.t(langTokens.admin.pictureRequired),
-    lastName: i18n.t(langTokens.admin.fieldCantBeEmpty),
-    firstName: i18n.t(langTokens.admin.fieldCantBeEmpty),
-    regionId: i18n.t(langTokens.admin.fieldCantBeEmpty),
-    cityId: i18n.t(langTokens.admin.fieldCantBeEmpty),
-    email: i18n.t(langTokens.admin.fieldCantBeEmpty),
-    work: i18n.t(langTokens.admin.fieldCantBeEmpty),
-    bio: i18n.t(langTokens.admin.fieldCantBeEmpty),
+    avatar: author?.avatar ? '' : i18n.t(langTokens.admin.pictureRequired),
```

Comment:
but why do we need to keep error messages in the state? what if we will calculate it dynamically with `useMemo`?

```
const errorMessages = useMemo(() => {
  const errors = {
    email: '',
    lastName: '',
    // ...
  };
  if (!author?.email) {
    errors[email] =  i18n.t(langTokens.admin.fieldCantBeEmpty),;
  } else if (value && !regex.validEmail.test(author.email)) {
      errors.email = i18n.t(langTokens.admin.wrongEmail);
  }
  .....
}, [author?.email, author?.bio, author?.region.id])
```

then we would not need to call `setErrorFields` just to update error messages. And keeping all the validation logic in single place may be beneficial - though, this block might grow in size, it's a typical case where for validation we need to check _several_ fields which depend on each other. E.g. just an synthetic example - for address block which is optional, we either keep all the fields empty, but if we provide street name, we must also provide a city ID. When validation in single place, it's easier to implement checks like that.

Reply:
I applied. If I understand correctly, this significantly reduced and improved the code.